whackaworm.js

```diff
@@ -1,123 +1,88 @@
-var imgArr;
-var hole1;
-var hole2;
-var hole3;
-var hole4;
-var hole5;
-var hole6;
-var hole7;
-var hole8;
-var hole9;
-var worm;
-var score;
-var randomNumber;
-var previousRandomNum;
+var holes, worm, hole, score, randomNum, previousRandomNum, timeout, eventFunctionName, numOfHoles;
 
-function init()
-{
-	score = 0;
-	previousRandomNum = 0;
-	randomNumber = 0;
+function init(){
+	initializeAssets();
 	cacheImages();
 	
-	addEvent(hole1, "click", randomize, false);
-	addEvent(hole2, "click", randomize, false);
-	addEvent(hole3, "click", randomize, false);
-	addEvent(hole4, "click", randomize, false);
-	addEvent(hole5, "click", randomize, false);
-	addEvent(hole6, "click", randomize, false);
-	addEvent(hole7, "click", randomize, false);
-	addEvent(hole8, "click", randomize, false);
-	addEvent(hole9, "click", randomize, false);
+	for(i=0; i < holes.length;i ++){
+		addEvent(holes[i], clickAddEventFunction, randomize, false);
+	}
 
 	play();
-	setTimeout("endGame()",60000);
-
-}
-
-function generateNum()
-{
-	randomNumber = Math.floor(Math.random()*10);
-	while(previousRandomNum == randomNumber || randomNumber == 9){
-		randomNumber = Math.floor(Math.random()*10);
-	}
+	setTimeout("endGame()", timeout);
 }
 
-function addEvent(objName, type, fnName, Cap) {
-
-  if(objName.attachEvent)
-  {
-    objName.attachEvent("on"+type,fnName);
-  }
-  else if(objName.addEventListener)
-  {
-    objName.addEventListener(type,fnName,Cap);
-  }
+function initializeAssets(){
+	numOfHoles = 9;
+	timeout = 60000;
+	clickAddEventFunction = "click";
+	score = 0;
+	previousRandomNum = 0;
+	randomNum = 0;
+	hole = "hole.png";
+	worm = "wormHoleImage.png";
+	holes = [];
 }
 
-function randomize(e)
-{
-	var source = e.target ||event.srcElement;
-	var img = source.src;
-
-	if(img.substring(img.lastIndexOf("/")+1) == worm){
-
-	score+=10;
-	play();
+function cacheImages(){
+    //Cache all the image tags
+	for(i = 1; i <= numOfHoles; i++){
```

Comment:
start i at 0, i < numOfHoles
then, "hole" + (i + 1).toString()

also, "hole" shouldn't be hardcoded